lang/java/reef-runtime-local/src/main/java/org/apache/reef/runtime/local/process/RunnableProcess.java

```diff
@@ -160,57 +141,72 @@ public void run() {
 
       // Launch the process
       try {
-        LOG.log(Level.FINEST, "Launching process \"{0}\"\nSTDERR can be found in {1}\nSTDOUT can be found in {2}",
-            new Object[]{this.id, errFile.getAbsolutePath(), outFile.getAbsolutePath()});
+
+        LOG.log(Level.FINEST,
+            "Launching process \"{0}\"\nSTDERR can be found in {1}\nSTDOUT can be found in {2}",
+            new Object[] {this.id, errFile.getAbsolutePath(), outFile.getAbsolutePath()});
+
         this.process = new ProcessBuilder()
             .command(this.command)
             .directory(this.folder)
             .redirectError(errFile)
             .redirectOutput(outFile)
             .start();
-        this.setState(State.RUNNING);
+
+        this.setState(RunnableProcessState.RUNNING);
         this.processObserver.onProcessStarted(this.id);
+
       } catch (final IOException ex) {
-        LOG.log(Level.SEVERE, "Unable to spawn process \"{0}\" wth command {1}\n Exception:{2}",
-            new Object[]{this.id, this.command, ex});
+        LOG.log(Level.SEVERE,
+            "Unable to spawn process \"{0}\" wth command {1}\n Exception:{2}",
+            new Object[] {this.id, this.command, ex});
       }
+
     } finally {
       this.stateLock.unlock();
     }
 
     try {
+
       // Wait for its completion
-      final int returnValue = process.waitFor();
+      LOG.log(Level.FINER, "Wait for process completion: {0}", this.id);
+      final int returnValue = this.process.waitFor();
       this.processObserver.onProcessExit(this.id, returnValue);
+
       this.stateLock.lock();
       try {
-        this.setState(State.ENDED);
+        this.setState(RunnableProcessState.ENDED);
         this.doneCond.signalAll();
       } finally {
         this.stateLock.unlock();
       }
-      LOG.log(Level.FINEST, "Process \"{0}\" returned {1}", new Object[]{this.id, returnValue});
+
+      LOG.log(Level.FINER, "Process \"{0}\" returned {1}", new Object[] {this.id, returnValue});
+
     } catch (final InterruptedException ex) {
-      LOG.log(Level.SEVERE, "Interrupted while waiting for the process \"{0}\" to complete. Exception: {2}",
-          new Object[]{this.id, ex});
+      LOG.log(Level.SEVERE,
+          "Interrupted while waiting for the process \"{0}\" to complete. Exception: {2}",
```

Comment:
Good catch! Sometimes even simple code reformatting helps to find bugs :)

**P.S.** I am done with the fixes. @tcNickolas could you please take another look? Thank you!